core/src/Streamly/Internal/Data/MutArray/Type.hs

```diff
@@ -2582,7 +2596,240 @@ byteEq :: MonadIO m => MutArray a -> MutArray a -> m Bool
 byteEq arr1 arr2 = fmap (EQ ==) $ byteCmp arr1 arr2
 
 -------------------------------------------------------------------------------
--- NFData
+-- Compact
+-------------------------------------------------------------------------------
+
+-- | Parser @pCompactChunksLE n@ coalesces adjacent arrays in the input stream
+-- only if the combined size would be less than or equal to n.
+--
+-- /Internal/
+{-# INLINE_NORMAL pCompactChunksLE #-}
+pCompactChunksLE ::
```

Comment:
Does `p` here signify parser?

Reply:
Can we move this to the parser module? `Chunks` is already in the name.

Reply:
If we want to keep it here, can we rename it to `compactChunksParserLE`? Although long it's explicit.
The prefix `p` is confusing.

Reply:
it's been renamed to pCompactLE in a later PR. adding Parser/Fold suffix may be verbose especially if we follow this convention in more places.